src/dash-playlist-loader.js

```diff
@@ -0,0 +1,173 @@
+import { EventTarget } from 'video.js';
+import mpdParser from 'mpd-parser';
+import {
+  setupMediaPlaylists,
+  resolveMediaGroupUris
+} from './playlist-loader';
+
+export default class DashPlaylistLoader extends EventTarget {
+  // DashPlaylistLoader must accept either a src url or a playlist because subsequent
+  // playlist loader setups from media groups will expect to be able to pass a playlist
+  // (since there aren't external URLs to media playlists with DASH)
+  constructor(srcUrlOrPlaylist, hls, withCredentials) {
+    super();
+
+    this.hls_ = hls;
+    this.withCredentials = withCredentials;
+
+    if (!srcUrlOrPlaylist) {
+      throw new Error('A non-empty playlist URL or playlist is required');
+    }
+
+    // initialize the loader state
+    if (typeof srcUrlOrPlaylist === 'string') {
+      this.srcUrl = srcUrlOrPlaylist;
+      this.state = 'HAVE_NOTHING';
+      return;
+    }
+
+    this.state = 'HAVE_METADATA';
+    this.started = true;
+    // we only should have one, so select it
+    this.media(srcUrlOrPlaylist);
+    this.trigger('loadedplaylist');
+  }
+
+  dispose() {
+    this.stopRequest();
+  }
+
+  stopRequest() {
+    if (this.request) {
+      const oldRequest = this.request;
+
+      this.request = null;
+      oldRequest.onreadystatechange = null;
+      oldRequest.abort();
+    }
+  }
+
+  media(playlist) {
```

Comment:
Would it be possible to get function comments here?  Mostly I'm asking as this has a few pieces of nuance (the playlist can be a string or object, and thus will be handled differently, as well as the fact that it is both a getter and a setter).